providers/src/airflow/providers/edge/worker_api/routes/worker.py

```diff
@@ -0,0 +1,195 @@
+        {
+            "state": "running",
+            "jobs_active": 3,
+            "queues": ["large_node", "wisconsin_site"],
+            "sysinfo": {
+                "concurrency": 4,
+                "airflow_version": "2.10.0",
+                "edge_provider_version": "1.0.0",
+            },
+        }
+    ],
+)
+_worker_queue_doc = Body(
+    title="Changes in worker queues",
+    description="Changes to be applied to current queues of worker",
+    examples=[{"new_queues": ["new_queue"], "remove_queues": ["old_queue"]}],
+)
+
+
+@worker_router.post(
+    "/{worker_name}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def register(
+    worker_name: Annotated[str, _worker_name_doc],
+    body: Annotated[WorkerStateBody, _worker_state_doc],
+    session: Annotated[Session, Depends(get_session)],
+) -> datetime:
+    """Register a new worker to the backend."""
+    _assert_version(body.sysinfo)
+    query = select(EdgeWorkerModel).where(EdgeWorkerModel.worker_name == worker_name)
+    worker: EdgeWorkerModel = session.scalar(query)
+    if not worker:
+        worker = EdgeWorkerModel(worker_name=worker_name, state=body.state, queues=body.queues)
+    worker.state = body.state
+    worker.queues = body.queues
+    worker.sysinfo = json.dumps(body.sysinfo)
+    worker.last_update = timezone.utcnow()
+    session.add(worker)
+    return worker.last_update
+
+
+@worker_router.patch(
+    "/{worker_name}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def set_state(
+    worker_name: Annotated[str, _worker_name_doc],
+    body: Annotated[WorkerStateBody, _worker_state_doc],
+    session: Annotated[Session, Depends(get_session)],
+) -> list[str] | None:
+    """Set state of worker and returns the current assigned queues."""
+    query = select(EdgeWorkerModel).where(EdgeWorkerModel.worker_name == worker_name)
+    worker: EdgeWorkerModel = session.scalar(query)
+    worker.state = body.state
+    worker.jobs_active = body.jobs_active
+    worker.sysinfo = json.dumps(body.sysinfo)
+    worker.last_update = timezone.utcnow()
+    session.commit()
+    Stats.incr(f"edge_worker.heartbeat_count.{worker_name}", 1, 1)
+    Stats.incr("edge_worker.heartbeat_count", 1, 1, tags={"worker_name": worker_name})
+    set_metrics(
+        worker_name=worker_name,
+        state=body.state,
+        jobs_active=body.jobs_active,
+        concurrency=int(body.sysinfo.get("concurrency", -1)),
+        free_concurrency=int(body.sysinfo["free_concurrency"]),
+        queues=worker.queues,
+    )
+    _assert_version(body.sysinfo)  #  Exception only after worker state is in the DB
+    return worker.queues
```

Comment:
This isn't listed as response in the endpoint metadata

Reply:
I do not understand what you mean. `_assert_version()` will raise HTTP 400 if not good, otherwise the return value is `list[str] | None` as in the signature. What do you think is missing?

Reply:
Ah, do you mean that "None" must declare HTTP 204 code?

Reply:
I attempted to add HTTP 204 as a code - but if the response is "in some cases" empty, then this is not accepted. Error from FastAPI is:
```
  File "/opt/airflow/providers/src/airflow/providers/edge/worker_api/routes/rpc_api.py", line 188, in <module>
    @rpc_api_router.post(
     ^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/airflow/api_fastapi/common/router.py", line 65, in decorator
    self.add_api_route(
  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 933, in add_api_route
    route = route_class(
            ^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 540, in __init__
    assert is_body_allowed_for_status_code(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Status code 204 must not have a response body
```

Reply:
aah right